im-sdk-core/src/main/java/com/easemob/im/server/api/attachment/download/Download.java

```diff
@@ -19,12 +19,12 @@ public Download(Context context) {
         Path local = FileSystem.choosePath(dir, filename);
         return Mono.<OutputStream>create(sink -> sink.success(FileSystem.open(local)))
                 .flatMap(out -> this.context.getHttpClient()
-                        .get()
-                        .uri(String.format("/chatfiles/%s", id))
-                        .response((rsp, buf) -> this.context.getErrorMapper().apply(rsp).thenMany(buf))
-                        .doOnNext(buf -> FileSystem.append(out, buf))
-                        .doFinally(sig -> FileSystem.close(out))
-                        .then())
+                        .flatMap(httpClient -> httpClient.get()
```

Comment:
HttpClient  -->  Mono<HttpClient>
so all the transformation stuff now must be wrapped within a flatMap